pkg/logs/decoder/line_generator.go

```diff
@@ -26,6 +28,19 @@ type LineGenerator struct {
 	endLineMatcher   EndLineMatcher
 	convertor        parser.Convertor
 	handlerScheduler LineHandlerScheduler
+	lineBuf          *lineBuffer
+}
+
+// NewLineGenerator creates a new instance of LineGenerator.
+func NewLineGenerator(maxDecodingLen int, inputChan chan *Input, endLineMatcher EndLineMatcher, convertor parser.Convertor, handlerScheduler LineHandlerScheduler) *LineGenerator {
```

Comment:
Same as in an another PR, here you rewrote all the decoding logic, why so ? I think it was possible to work on top of the previous decoder and improve it to work with your new design.

Reply:
I do recommend you to checkout this PR and run the tests, you will see the decoding logic is the same as before.

Reply:
I have no doubt that it works, just afraid of side effects